wowchemy/layouts/slides/baseof.html

```diff
@@ -20,55 +21,176 @@
 
   <title>{{ .Title }} | {{ site.Title }}</title>
 
-  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/css/reveal.min.css">
+  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/reveal.min.css">
   {{- $theme := $.Param "slides.theme" | default "black" -}}
-  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/css/theme/{{ $theme }}.min.css">
+  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/theme/{{ $theme }}.min.css" id="theme" />
 
   {{- $highlight_style := $.Param "slides.highlight_style" | default "dracula" -}}
-  {{ printf "<link rel=\"stylesheet\" href=\"%s\" crossorigin=\"anonymous\">" (printf $css.highlight.url $css.highlight.version $highlight_style) | safeHTML }}
+  {{ printf "<link rel=\"stylesheet\" href=\"%s\" crossorigin=\"anonymous\" id=\"highlight-theme\">" (printf $css.highlight.url $css.highlight.version $highlight_style) | safeHTML }}
 
-  {{ $css := resources.Get "css/reveal.css" }}
   {{ $css_custom := resources.Get "css/reveal_custom.css" }}
-  {{ $style := slice $css $css_custom | resources.Concat "css/reveal_custom.css" | resources.Minify }}
+  {{ $style := slice $css_custom | resources.Concat "css/reveal_custom.css" | resources.Minify }}
   <link rel="stylesheet" href="{{ $style.RelPermalink }}">
 
-  <!-- Printing and PDF exports -->
-  <script>
-    var link = document.createElement( 'link' );
-    link.rel = 'stylesheet';
-    link.type = 'text/css';
-    link.href = window.location.search.match( /print-pdf/gi ) ? '{{ $cdn_url_reveal }}/css/print/pdf.css' : '{{ $cdn_url_reveal }}/css/print/paper.css';
-    document.getElementsByTagName( 'head' )[0].appendChild( link );
-  </script>
-
 </head>
 <body>
 
   {{ block "main" . }}{{ end }}
 
-  <script src="{{ $cdn_url_reveal }}/js/reveal.min.js"></script>
+  <script src="{{ $cdn_url_reveal }}/reveal.min.js"></script>
+
+  <script src="{{ $cdn_url_reveal }}/plugin/markdown/markdown.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/highlight/highlight.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/notes/notes.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/search/search.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/math/math.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/zoom/zoom.js"></script>
+
+<!-- 3rd part plugins -->
+
+  <script src="/js/vendor/reveal.js/plugin/menu/menu.js"></script>
```

Comment:
Why are you suggesting to load some Reveal-related JS assets from CDN and some locally?

Reply:
> Why are you suggesting to load some Reveal-related JS assets from CDN and some locally?

Both the `menu` and the `chalkboard` plugins are not built-in plugins are therefore are not available from a CDN. Ideally they could be optional (to load) but I really think they add some very interesting value.

Reply:
exactly what @rodrigoalcarazdelaosa said.

There is even a request to simplify adding reveal plugins ... might look into it as it seems very doable

will be a busy week .. expect delays

Reply:
BTW, I'd prefer for the `menu` plugin to only show the menu by default, not also the possibility to select a different theme.

Reply:
> > Why are you suggesting to load some Reveal-related JS assets from CDN and some locally?
> 
> Both the `menu` and the `chalkboard` plugins are not built-in plugins and therefore are not available from a CDN.

I've just done a quick search and the plugins you suggested do seem to be published in one of the popular JS CDNs: https://www.jsdelivr.com/package/npm/reveal.js-plugins

It would facilitate reviewing (and hopefully significantly reduce the number of new/changed files from **70**) if we could at least refactor this PR to only use CDN for assets.

Reply:
As for plugins, I suggest to consider including some plugins which include common presentation functionality such as Speaker Notes, Menu, and Chalkboard by default.

In the future, we can consider adding something like `slide_plugins` in front matter to include additional plugins.

As for configuring slides and plugins, let's consider renaming `slides` to `slide_options` so that the namespace doesn't clash with the use of `slides` elsewhere for linking slides to pages. Then we can consider having sub-options under that for each plugin, similar to the front matter structure used in Rmarkdown: https://bookdown.org/yihui/rmarkdown/reveal-js-plugins.html

Reply:
renaming all options to use lowercase and underscore names instead of camel case. I hope we can keep this consistent for the future.

I like the blowdown approach for plugins and should be the way forward.

speaker notes is there by default. Its now shipped with reveal and enabled by default along with the following:

  <script src="{{ $cdn_url_reveal }}/reveal.min.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/markdown/markdown.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/highlight/highlight.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/notes/notes.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/search/search.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/math/math.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/zoom/zoom.js"></script>
chalkboard is a small plugin and the only version available on can is an old one that is not compatible with reveal4. So I kets the files in the static dir.

I did exclude menu plug in file since I was able to successfully add it from cdn. It seems to work perfectly, but the following error can be seen in the console:

SyntaxError: import.meta is only valid inside modules.

Not sure how that might impact slides, so please test and let me know.

in the slide front matter, the slides: namespace was already there to configure slide themes and transitions. It isn't something that I added. I just switched from hard coded configuration into a more dynamic one where the dictionary is read with all the configuration and the options passed to reveal and all other plugin.

Only challenge is defaults. I'll have to create an object in Wowchemy-slides.js which include some of the default values when the options are not set.

I think keeping it as slides: is cleaner, unless you can think of an instance where there indeed is a conflict. It will ensure that older slides will remain compatible with the new update.

I will add defaults to enable menu and chalkboard, but will keep the option to disable them in case someone runs into a compatibility issue with them on their slides.

pushing an update soon

Reply:
im also moving the menu and chalkboard options under reveal_options since they are plugins

Reply:
Thanks, great progress on this!

As for local assets, please continue removing the remaining Reveal-related local assets (Chalkboard) from this PR and consider requesting the author to publish the Reveal v4 compatible version to the CDN if necessary.

Reply:
@dralmarzouq in case you missed this comment ^

Reply:
removed chalkboard until it is hosted on cdn .. I can host on my own website if needed
only menu is now available by default.
online classes will not be the same without chalkboard

Reply:
> removed chalkboard until it is hosted on cdn .. I can host on my own website if needed
> only menu is now available by default.
> online classes will not be the same without chalkboard

Hoping that [rajgoel](https://github.com/rajgoel) will update it in the CDN soon 💪🏼

Reply:
@rodrigoalcarazdelaosa would definitely help if you know him and can encourage him to do so ... I somehow doubt he will look at the ticket I submitted. The repo has been inactive for a while now.

Reply:
> @rodrigoalcarazdelaosa would definitely help if you know him and can encourage him to do so ... I somehow doubt he will look at the ticket I submitted. The repo has been inactive for a while now.

I'm afraid I don't have any relationship with him but I have given a thumbs up in [your issue](https://github.com/rajgoel/reveal.js-plugins/issues/106). I'd change the title to _consider updating_ instead of _consider hosting_, as the package is hosted but not updated. If hosting the plugin locally works, it's just a matter of updating the plugin in the CDN, which shouldn't take much effort right?

Reply:
I did try using jsdeliver github mirroring cdn ... did not work for some reason.
You might have better success than me.
Worked only when hosted locally